src/lib/components/ui/FilterBuilder.svelte

```diff
@@ -37,40 +38,56 @@
 
 	// Get field type for the selected field
 	$: selectedFieldSchema = $fieldSchema[newFilterField] || null;
-	$: selectedFieldType = selectedFieldSchema?.type || 'string';
-	$: selectedFieldUiType = selectedFieldSchema?.ui_type || 'short_text';
+	$: selectedFieldType =
+		getMappingFieldType(newFilterField) || selectedFieldSchema?.type || 'string';
+	$: selectedFieldUiType =
+		getMappingFieldUiType(newFilterField) || selectedFieldSchema?.ui_type || 'short_text';
 	$: isSelectField = selectedFieldUiType === 'select';
-	$: fieldOptions = isSelectField ? selectedFieldSchema?.options || [] : [];
-
-	// Force equals operator for select fields
-	$: if (isSelectField) {
+	$: fieldOptions =
+		getMappingFieldOptions(newFilterField) ||
+		(isSelectField ? selectedFieldSchema?.options || [] : []);
+
+	// Force equals operator for select fields (but allow operators for mapping fields)
+	// Also force equals operator for has_mappings field
+	$: if (
+		(isSelectField && !['has_mappings', 'mapping_status'].includes(newFilterField)) ||
+		newFilterField === 'has_mappings' ||
+		newFilterField === 'mapping_status'
+	) {
```

Comment:
```suggestion
	$: if (isSelectField) {
```

I think since the other two are select fields this can be simplified to just a single if again.